### PR TITLE
DateTime validator support for trailing data

### LINKED ISSUE #37094: DateTime constraint does not support trailing data
**Description**  
The DateTime constraint can not deal with explicitly ignored trailing data. The validation will fail even though the value matches the specified format.

**Example**  
```        
$validator = Validation::createValidator();
$violations = $validator->validate('2020-01-01 00:00:00', new DateTime('Y-m-d+'));
```
The code above will create a violation with the message "This value is not a valid datetime."
The "+" in the format allows trailing data (see https://www.php.net/manual/en/datetime.createfromformat.php) BUT DateTime::getLastErrors() will return a warning:
```
 array:4 [
  "warning_count" => 1
  "warnings" => array:1 [
    10 => "Trailing data"
  ]
  "error_count" => 0
  "errors" => []
]
```
I suggest that in case the format ends with a "+" the warning should be ignored and don't cause a constraint violation.

### PR DESCRIPTION
**Pull request by stefankleff**
| Q             | A
| ------------- | ---
| Branch?       | 3.4
| Bug fix?      | yes
| New feature?  | no 
| Deprecations? | no 
| Tickets       | Fix #37094
| License       | MIT

### CODE REVIEW

File: src/Symfony/Component/Validator/Constraints/DateTimeValidator.php [hunk 1]
```diff
@@ -68,6 +68,8 @@ public function validate($value, Constraint $constraint)
                     ->setParameter('{{ value }}', $this->formatValue($value))
                     ->setCode(DateTime::INVALID_TIME_ERROR)
                     ->addViolation();
+            } elseif ('Trailing data' === $warning && '+' === substr($constraint->format, -1)) {
```
Comment (xabbuh) [inline src/Symfony/Component/Validator/Constraints/DateTimeValidator.php:71]: Could there be cases with more than one warning where don't want to ignore the other ones?

Comment (nicolas-grekas) [reply src/Symfony/Component/Validator/Constraints/DateTimeValidator.php:71]: @stefankleff any idea? Could you please have a look?

Comment (fabpot) [reply src/Symfony/Component/Validator/Constraints/DateTimeValidator.php:71]: @stefankleff Have you time to have a look at this?

Comment (stefankleff) [reply src/Symfony/Component/Validator/Constraints/DateTimeValidator.php:71]: Sorry, not yet. But I'll try within this week.

Comment (nicolas-grekas) [reply src/Symfony/Component/Validator/Constraints/DateTimeValidator.php:71]: friendly ping

Comment (stefankleff) [reply src/Symfony/Component/Validator/Constraints/DateTimeValidator.php]: The case is valid, that we have a "+" in the format and we don't want to ignore the other warnings.
I couldn't find a better way than remove the warning from the list of warnings.
I also looked in the underlying c-code and I couldn't find any similar cases to trailing data.

**Comment by fabpot** [comment]
Can you apply the patch suggested by fabbot?

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @stefankleff.

**Pull request merged**